gateway/fedimint-lightning/src/ldk.rs

```diff
@@ -141,10 +141,11 @@ impl GatewayLdkClient {
                 let host = server_url
```

Comment:
> I'm not sure how to enforce that we check this in CI. This was only discovered once running in an environment outside of devimint. Open to suggestions.

What do you think of pulling the logic used to construct `server_url` into a separate function and adding unit tests for the different types of URLs? Feel free to punt if you don't think that's capturing what you were hoping to test.

Reply:
Done